### ISSUE TITLE
Emphasize use of args to configure arguments for kubernetes client authentication plugins

### ISSUE THREAD

**Issue by divijan** [kubernetes/website#25961]
There are a lot of undocumented gotchas (some of them are probably bugs) about using kubeconfig for configuring client-go authentication under Windows. 
First, there is no information on which characters in kubeconfig are legal. I found out that Windows environment variables enclosed in `%` should be referenced inside single quotes. Double quotes don't work.
Second, there has to be a single executable referenced in `command: `, nothing like `python <path-to-script>` works: kubernetes complains it cannot find such an executable. It could be a .bat file.

**Comment by sftim** [comment]
If these _are_ bugs @divijan , https://github.com/kubernetes/kubectl/ is the best place to report them.

If you want to know what characters are legal in Kubeconfig, you can look at https://yaml.org/
(maybe the documentation could link there).


The page at https://k8s.io/docs/reference/access-authn-authz/authentication/ looks fine in the regard; its main topic is Kubernetes authentication.

**Comment by divijan** [comment]
OK, in that case I would suggest adding a note in **client-go credential plugins \ Configuration** section that command must not contain spaces.

**Comment by sftim** [comment]
@divijan if you had a credential plugin with a space in the name of the binary I think it would work. Have I got that right?

We could explain that you use `args` to set command line arguments, just next to the line about `command`.

/retitle Emphasize use of args to configure arguments for kubernetes client authentication plugins

**Comment by divijan** [comment]
@sftim yes, you are correct. However, delving deeper into my case, I had a python script that did authentication, and I could not put `python <script name>` into `command:`, so I had to enclose it into a .bat file. But now during this discussion I tried putting `<script name>` into `args:` before arguments for the script itself and this way script couldn't read the file. It worked when script invocation was contained in a .bat. Any comments on this?
